nemo/collections/multimodal/speech_llm/models/modular_s2s_models.py

```diff
@@ -283,17 +283,23 @@ def restore_from_pretrained_models(
         cfg: Optional[Union[OmegaConf, str]] = None,
         trainer: Optional[Trainer] = None,
     ):
-        model = super().restore_from_pretrained_models(cfg, trainer)
+
+        model, codec_model, asr_model, mos_model = super().restore_from_pretrained_models(cfg, trainer)
+        model, codec_model, asr_model = super().restore_from_pretrained_models(cfg, trainer)
         if cfg.model.get('salm_model_path') is not None:
             torch_state_dict = torch.load(cfg.model.get('salm_model_path'))['state_dict']
             model.setup_complete = False
             model.load_state_dict(torch_state_dict, strict=False)
             logging.info(f"loading from {cfg.model.get('salm_model_path')}: {torch_state_dict.keys()}")
 
         model.padded_vocab_size = cfg.model.s2s_vocab_size
-        model.model.extend_embedding(model.padded_vocab_size)
+        model.model.module.extend_embedding(model.padded_vocab_size)
```

Comment:
this will only work for amp
can u wrap these using if self.megatron_amp_O2 and keep the original code for False case
same for the following

Reply:
fixed